x-pack/plugins/security_solution/public/detection_engine/rule_management/logic/bulk_actions/use_bulk_export.ts

```diff
@@ -6,44 +6,43 @@
  */
 
 import { useCallback } from 'react';
-import type { BulkActionResponse, BulkActionSummary } from '..';
 import { BulkAction } from '../../../../../common/detection_engine/rule_management/api/rules/bulk_actions/request_schema';
-import type { HTTPError } from '../../../../../common/detection_engine/types';
 import type { UseAppToasts } from '../../../../common/hooks/use_app_toasts';
 import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
 import { downloadBlob } from '../../../../common/utils/download_blob';
 import * as i18n from '../../../../detections/pages/detection_engine/rules/translations';
+import { useRulesTableContextOptional } from '../../../rule_management_ui/components/rules_table/rules_table/rules_table_context';
 import { getExportedRulesCounts } from '../../../rule_management_ui/components/rules_table/helpers';
-import type { RulesTableActions } from '../../../rule_management_ui/components/rules_table/rules_table/rules_table_context';
 import { useBulkExportMutation } from '../../api/hooks/use_bulk_export_mutation';
-import { getErrorToastContent, getSuccessToastContent } from './translations';
+import { showBulkErrorToast } from './show_bulk_error_toast';
+import { showBulkSuccessToast } from './show_bulk_success_toast';
+import type { QueryOrIds } from '../../api/api';
 
-interface RulesBulkActionArgs {
-  visibleRuleIds?: string[];
-  search: { query: string } | { ids: string[] };
-  setLoadingRules?: RulesTableActions['setLoadingRules'];
-}
-
-export const useBulkExport = () => {
+export function useBulkExport() {
   const toasts = useAppToasts();
   const { mutateAsync } = useBulkExportMutation();
+  const rulesTableContext = useRulesTableContextOptional();
+  const setLoadingRules = rulesTableContext?.actions.setLoadingRules;
 
   const bulkExport = useCallback(
-    async ({ visibleRuleIds = [], setLoadingRules, search }: RulesBulkActionArgs) => {
+    async (queryOrIds: QueryOrIds) => {
       try {
-        setLoadingRules?.({ ids: visibleRuleIds, action: BulkAction.export });
-        return await mutateAsync(search);
+        setLoadingRules?.({
+          ids: queryOrIds.ids ?? [],
+          action: BulkAction.export,
+        });
+        return await mutateAsync(queryOrIds);
```

Comment:
Once the user clicked any of the bulk actions, we should "lock" the table: the user shouldn't be able to change the filters and the rule selection until the bulk action is sent or canceled. If we don't do this like in the 2nd video, there's a chance that the user could make some changes in the table which would lead to different arguments of the dry run request and the final request. Which would be a bug.

I think we should fix this. @maximpn could we fix this for all the bulk actions in a follow-up PR after merging this one? @xcrzx could you please write up a ticket for this issue?

Reply:
Added a separate ticket: https://github.com/elastic/kibana/issues/144264